truffle/README.md

```diff
@@ -0,0 +1,226 @@
+   sudo sh get-docker.sh
+   ```
+
+2. Configure Docker settings.
+
+   ```sh
+   sudo usermod -a -G docker `whoami`
+   ```
+   
+3. Install [Docker Compose](https://docs.docker.com/compose/install/).
+  
+   ```sh
+   sudo apt install -y docker-compose
+   ```
+
+4. Install [Node.js](https://nodejs.org/en/). The version in the OS repository is 
+  out of date, so we'll get the package from a different source. 
+  
+   ```sh
+   curl -sL https://deb.nodesource.com/setup_12.x -o nodesource_setup.sh
+   sudo bash nodesource_setup.sh
+   sudo apt install -y nodejs
+   ```
+   
+5. Install the Node.js packages we need. [See here](https://github.com/sindresorhus/guides/blob/main/npm-global-without-sudo.md)
+   if you'd rather install these packages without root permissions.
+   ```sh   
+   sudo npm install -g yarn truffle ganache-cli
+   ```
+   
+6. Log out and log back in to complete the Docker installation (required).
+
+### Start an Optimistic Ethereum Node
+
+This process downloads, compiles, and builds an Optimistic Ethereum network. Note that it takes a long time.
+
+1. Clone the [Optimism monorepo](https://github.com/ethereum-optimism/optimism).
+
+   ```sh
+   git clone https://github.com/ethereum-optimism/optimism.git
+   cd optimism
+   ```
+   
+2. Build the Optimistic Ethereum software.   
+   
+   ```sh
+   yarn install
+   yarn build
+   ```
+   
+3. Build the Docker containers
+
+   ```sh
+   cd ops
+   export COMPOSE_DOCKER_CLI_BUILD=1
+   export DOCKER_BUILDKIT=1
+   docker-compose build && echo Build complete
+   ```
+
+The build process is time consuming, and you do not need to wait for it to finish before you continue the tutorial.
+I will note the point in the tutorial where you need to have a running Optimistic Ethereum Node. Hopefully it will
+be finished by then (you will know when the build process is done because you'll see a **Build complete** message).
+
+4. Once the build process is finally done, start the Optimistic Ethereum node:
+
+   ```sh
+   docker-compose up
+   ```
+
+5. To see when the Optimistic Ethereum node starts, run (in a separate terminal):
+   
+   ```sh
+   ~/optimism/ops/scripts/wait-for-sequencer.sh
+   ```
+
+
+
+
+## Migrate a Dapp to Optimistic Ethereum
+
+Now that we have Optimistic Ethereum running, it is time to run a distributed application (dapp) on it.
+
+**Note:** If you don't need the explanations and just want to see running code, 
+[click here](https://github.com/ethereum-optimism/optimism-tutorial/). The 
+`/truffle/dapp` directory
+is just an `npm install` away from being a working example.
+
+### Get a Sample Application
+
+The easiest way is to start with a sample application. 
+
+1. Open a second command line terminal
+2. Run `truffle`, the development environment we use in this tutorial, to 
+   [unbox an Optimistic Ethernet application](https://www.trufflesuite.com/boxes/optimism).
+   ```sh
+   mkdir dapp
+   cd dapp
+   truffle unbox optimism
```

Comment:
This works although it somewhat "hides" the truffle configuration necessary to get contracts to compile for the ovm in truffle, i.e. [the box](https://github.com/truffle-box/optimism-box#new-configuration-file) already contains the config and settings necessary whereas if we compare to the hardhat tutorial, there the sample we use is non-optimism specific and therefore it is clear which settings need to be added (https://github.com/ethereum-optimism/optimism-tutorial/tree/main/hardhat#migrate-the-sample-app-to-optimistic-ethereum).

Reply:
With the box available from truffle (and going to stay available indefinitely), I think we don't need to worry about the settings here the way we did with hardhat, so it's OK to skip it. 

At some point there should be a reference for configuration file settings, but that's a relatively low priority IMAO.

Reply:
I changed my mind - I'm addition a reference for `truffle-config.ovm.js` into the tutorial. Otherwise it does feel too easy.